File: Archiver/app_code/clsDIRECTORY.vb
Imports Microsoft.VisualBasic
Imports System.Configuration
Imports System
Imports System.Data.SqlClient
Imports System.Collections
Imports System.IO
Imports System.Data.Sql
 
Public Class clsDIRECTORY

    '** DIM the selected table columns 
    Dim DBARCH As New clsDatabaseARCH
    Dim DMA As New clsDma
    Dim UTIL As New clsUtility
    Dim LOG As New clsLogging


    Dim UserID As String = ""
    Dim IncludeSubDirs As String = ""
    Dim FQN As String = ""
    Dim DB_ID As String = ""
    Dim VersionFiles As String = ""
    Dim ckMetaData As String = ""
    Dim ckPublic As String = ""
    Dim ckDisableDir As String = ""
    Dim QuickRefEntry As String = "0"
    Dim OcrDirectory As String = ""
    Dim OcrPdf As String = ""
    Dim ArchiveBit As Boolean = False
    Dim DeleteOnArchive As String = ""

    'DeleteOnArchive
    '** Generate the SET methods 
    Public Sub setDeleteOnArchive(ByRef val As String)
        DeleteOnArchive = val
    End Sub
    Public Sub setSkipIfArchiveBit(ByRef val As String)
        If val.ToUpper.Equals("TRUE") Then
            ArchiveBit = True
        Else
            ArchiveBit = False
        End If
    End Sub
    Public Sub setOcrPdf(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Userid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        OcrPdf = val
    End Sub
    Public Sub setOcrDirectory(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Userid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        OcrDirectory = val
    End Sub
    Public Sub setUserid(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Userid' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        UserID = val
    End Sub
    Public Sub setQuickRefEntry(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        QuickRefEntry = val
    End Sub
    Public Sub setIncludesubdirs(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        IncludeSubDirs = val
    End Sub

    Public Sub setFqn(ByRef val As String)
        If Len(val) = 0 Then
            MessageBox.Show("SET: Field 'Fqn' cannot be NULL.")
            Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        FQN = val
    End Sub

    Public Sub setDb_id(ByRef val As String)
        If Len(val) = 0 Then
            val = "ECM.Library"
            'messagebox.show("SET: Field 'Db_id' cannot be NULL.")
            'Return
        End If
        val = UTIL.RemoveSingleQuotes(val)
        DB_ID = val
    End Sub

    Public Sub setVersionfiles(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        VersionFiles = val
    End Sub

    Public Sub setCkmetadata(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        ckMetaData = val
    End Sub

    Public Sub setCkpublic(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        ckPublic = val
    End Sub

    Public Sub setCkdisabledir(ByRef val As String)
        val = UTIL.RemoveSingleQuotes(val)
        ckDisableDir = val
    End Sub



    '** Generate the GET methods 
    Public Function getUserid() As String
        If Len(UserID) = 0 Then
            MessageBox.Show("GET: Field 'Userid' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(UserID)
    End Function

    Public Function getIncludesubdirs() As String
        If IncludeSubDirs.ToUpper.Equals("TRUE") Then
            IncludeSubDirs = "Y"
        End If
        If IncludeSubDirs.ToUpper.Equals("FALSE") Then
            IncludeSubDirs = "N"
        End If
        If IncludeSubDirs.ToUpper.Equals("1") Then
            IncludeSubDirs = "Y"
        End If
        If IncludeSubDirs.ToUpper.Equals("0") Then
            IncludeSubDirs = "N"
        End If

        Return UTIL.RemoveSingleQuotes(IncludeSubDirs)
    End Function

    Public Function getFqn() As String
        If Len(FQN) = 0 Then
            MessageBox.Show("GET: Field 'Fqn' cannot be NULL.")
            Return ""
        End If
        Return UTIL.RemoveSingleQuotes(FQN)
    End Function

    Public Function getDb_id() As String
        If Len(DB_ID) = 0 Then
            'messagebox.show("GET: Field 'Db_id' cannot be NULL.")
            Return "ECM.Library"
        End If
        Return UTIL.RemoveSingleQuotes(DB_ID)
    End Function

    Public Function getVersionfiles() As String
        If VersionFiles.ToUpper.Equals("TRUE") Then
            VersionFiles = "Y"
        End If
        If VersionFiles.ToUpper.Equals("FALSE") Then
            VersionFiles = "N"
        End If
        If VersionFiles.ToUpper.Equals("1") Then
            VersionFiles = "Y"
        End If
        If VersionFiles.ToUpper.Equals("0") Then
            VersionFiles = "N"
        End If

        Return UTIL.RemoveSingleQuotes(VersionFiles)
    End Function

    Public Function getCkmetadata() As String
        If ckMetaData.ToUpper.Equals("TRUE") Then
            ckMetaData = "Y"
        End If
        If ckMetaData.ToUpper.Equals("FALSE") Then
            ckMetaData = "N"
        End If
        If ckMetaData.ToUpper.Equals("1") Then
            ckMetaData = "Y"
        End If
        If ckMetaData.ToUpper.Equals("0") Then
            ckMetaData = "N"
        End If
        Return UTIL.RemoveSingleQuotes(ckMetaData)
    End Function

    Public Function getCkpublic() As String
        If ckPublic.ToUpper.Equals("TRUE") Then
            ckPublic = "Y"
        End If
        If ckPublic.ToUpper.Equals("FALSE") Then
            ckPublic = "N"
        End If
        If ckPublic.ToUpper.Equals("1") Then
            ckPublic = "Y"
        End If
        If ckPublic.ToUpper.Equals("0") Then
            ckPublic = "N"
        End If

        Return UTIL.RemoveSingleQuotes(ckPublic)
    End Function

    Public Function getCkdisabledir() As String
        Return UTIL.RemoveSingleQuotes(ckDisableDir)
    End Function



    '** Generate the Required Fields Validation method 
    Public Function ValidateReqData() As Boolean
        If UserID.Length = 0 Then Return False
        If FQN.Length = 0 Then Return False
        If DB_ID.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the Validation method 
    Public Function ValidateData() As Boolean
        If UserID.Length = 0 Then Return False
        If FQN.Length = 0 Then Return False
        If DB_ID.Length = 0 Then Return False
        Return True
    End Function


    '** Generate the INSERT method 
    Public Function Insert() As Boolean

        If QuickRefEntry.Length = 0 Then
            QuickRefEntry = "0"
        End If
        If OcrDirectory.Length = 0 Then
            MessageBox.Show("OcrDirectory must be set to either Y or N")
        End If

        FQN = FQN.Replace("'", "`")

        Dim b As Boolean = False
        Dim s As String = ""
        s = s + " INSERT INTO Directory("
        s = s + "UserID,"
        s = s + "IncludeSubDirs,"
        s = s + "FQN,"
        s = s + "DB_ID,"
        s = s + "VersionFiles,"
        s = s + "ckMetaData,"
        s = s + "ckPublic,"
        s = s + "ckDisableDir, QuickRefEntry, OcrDirectory, OcrPdf, ArchiveSkipBit, DeleteOnArchive) values ("
        s = s + "'" + UserID + "'" + ","
        s = s + "'" + IncludeSubDirs + "'" + ","
        s = s + "'" + FQN + "'" + ","
        s = s + "'" + DB_ID + "'" + ","
        s = s + "'" + VersionFiles + "'" + ","
        s = s + "'" + ckMetaData + "'" + ","
        s = s + "'" + ckPublic + "'" + ","
        s = s + "'" + ckDisableDir + "'" + ","
        s = s + "'" + QuickRefEntry + "'" + ","
        s = s + "'" + OcrDirectory + "'" + ","
        s = s + "'" + OcrPdf + "'" + ","
        If ArchiveBit = True Then
            s = s + "1" + ","
        Else
            s = s + "0" + ","
        End If
        s = s + "'" + DeleteOnArchive + "')"

        Return DBARCH.ExecuteSqlNewConn(s, False)

    End Function


    '** Generate the UPDATE method 
    Public Function Update(ByVal WhereClause As String, ByVal OcrDirectory As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If OcrDirectory.Length = 0 Then
            MessageBox.Show("OcrDirectory must be set to either Y or N")
        End If

        If Len(WhereClause) = 0 Then Return False

        Dim ckDisableDir As String = getCkdisabledir()
        If ckDisableDir.ToUpper = "TRUE" Then
            ckDisableDir = "Y"
        End If
        If ckDisableDir.ToUpper = "FALSE" Then
            ckDisableDir = "N"
        End If
        If ckDisableDir.ToUpper = "1" Then
            ckDisableDir = "Y"
        End If
        If ckDisableDir.ToUpper = "0" Then
            ckDisableDir = "N"
        End If

        s = ""
        s = s + " update Directory set "
        s = s + "UserID = '" + getUserid() + "'" + ", "
        s = s + "IncludeSubDirs = '" + getIncludesubdirs() + "'" + ", "
        s = s + "FQN = '" + getFqn() + "'" + ", "
        s = s + "VersionFiles = '" + getVersionfiles() + "'" + ", "
        s = s + "ckMetaData = '" + getCkmetadata() + "'" + ", "
        s = s + "ckPublic = '" + getCkpublic() + "'" + ", "
        s = s + "ckDisableDir = '" + ckDisableDir + "',"
        s = s + "OcrDirectory = '" + OcrDirectory + "',"
        s = s + "OcrPdf = '" + OcrPdf + "',"
        If ArchiveBit = True Then
            s = s + "ArchiveSkipBit = 1, "
        Else
            s = s + "ArchiveSkipBit = 0, "
        End If
        s = s + "DeleteOnArchive = '" + DeleteOnArchive + "'"


        WhereClause = " " + WhereClause
        s = s + WhereClause
        Return DBARCH.ExecuteSqlNewConn(s, False)
    End Function


    '** Generate the SELECT method 
    Public Function SelectRecs() As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "UserID,"
        s = s + "IncludeSubDirs,"
        s = s + "FQN,"
        s = s + "DB_ID,"
        s = s + "VersionFiles,"
        s = s + "ckMetaData,"
        s = s + "ckPublic,"
        s = s + "ckDisableDir "
        s = s + " FROM Directory"

        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the Select One Row method 
    Public Function SelectOne(ByVal WhereClause As String) As SqlDataReader
        Dim b As Boolean = False
        Dim s As String = ""
        Dim rsData As SqlDataReader
        s = s + " SELECT "
        s = s + "UserID,"
        s = s + "IncludeSubDirs,"
        s = s + "FQN,"
        s = s + "DB_ID,"
        s = s + "VersionFiles,"
        s = s + "ckMetaData,"
        s = s + "ckPublic,"
        s = s + "ckDisableDir "
        s = s + " FROM Directory"
        s = s + WhereClause

        Dim CS As String = DBARCH.setConnStr()     'DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
        Return rsData
    End Function


    '** Generate the DELETE method 
    Public Function Delete(ByVal WhereClause As String) As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        If Len(WhereClause) = 0 Then Return False

        WhereClause = " " + WhereClause

        s = " Delete from Directory"
        s = s + WhereClause

        b = DBARCH.ExecuteSqlNewConn(s, False)
        Return b

    End Function


    '** Generate the Zeroize Table method 
    Public Function Zeroize() As Boolean
        Dim b As Boolean = False
        Dim s As String = ""

        s = s + " Delete from Directory"

        b = DBARCH.ExecuteSqlNewConn(s, False)
        Return b

    End Function


    '** Generate Index Queries 
    Public Function cnt_PKII2(ByVal FQN As String, ByVal UserID As String) As Integer

        Dim B As Integer = 0
        Dim TBL As String = "Directory"
        Dim WC As String = "Where FQN = '" + FQN + "' and   UserID = '" + UserID + "'"

        B = DBARCH.iGetRowCount(TBL, WC)

        Return B
    End Function     '** cnt_PKII2

    '** Generate Index ROW Queries 
    Public Function getRow_PKII2(ByVal FQN As String, ByVal UserID As String) As SqlDataReader

        Dim rsData As SqlDataReader = Nothing
        Dim TBL As String = "Directory"
        Dim WC As String = "Where FQN = '" + FQN + "' and   UserID = '" + UserID + "'"

        rsData = DBARCH.GetRowByKey(TBL, WC)

        If rsData.HasRows Then
            Return rsData
        Else
            Return Nothing
        End If
    End Function     '** getRow_PKII2

    ''' Build Index Where Caluses 
    '''
    Public Function wc_PKII2(ByVal FQN As String, ByVal UserID As String) As String

Dim WC AS String  = "Where FQN = '" + FQN + "' and   UserID = '" + UserID + "'" 

        Return WC
    End Function     '** wc_PKII2
End Class

File: Archiver/app_code/clsDIRECTORY.cs
using global::System.Data.SqlClient;
using System.Windows.Forms;
using global::Microsoft.VisualBasic;
using MODI;

namespace EcmArchiver
{
    public class clsDIRECTORY
    {

        // ** DIM the selected table columns 
        private clsDatabaseARCH DBARCH = new clsDatabaseARCH();
        private clsDma DMA = new clsDma();
        private clsUtility UTIL = new clsUtility();
        private clsLogging LOG = new clsLogging();
        private string UserID = "";
        private string IncludeSubDirs = "";
        private string FQN = "";
        private string DB_ID = "";
        private string VersionFiles = "";
        private string ckMetaData = "";
        private string ckPublic = "";
        private string ckDisableDir = "";
        private string QuickRefEntry = "0";
        private string OcrDirectory = "";
        private string OcrPdf = "";
        private bool ArchiveBit = false;
        private string DeleteOnArchive = "";

        // DeleteOnArchive
        // ** Generate the SET methods 
        public void setDeleteOnArchive(ref string val)
        {
            DeleteOnArchive = val;
        }

        public void setSkipIfArchiveBit(ref string val)
        {
            if (val.ToUpper().Equals("TRUE"))
            {
                ArchiveBit = true;
            }
            else
            {
                ArchiveBit = false;
            }
        }

        public void setOcrPdf(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Userid' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            OcrPdf = val;
        }

        public void setOcrDirectory(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Userid' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            OcrDirectory = val;
        }

        public void setUserid(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Userid' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            UserID = val;
        }

        public void setQuickRefEntry(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            QuickRefEntry = val;
        }

        public void setIncludesubdirs(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            IncludeSubDirs = val;
        }

        public void setFqn(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                MessageBox.Show("SET: Field 'Fqn' cannot be NULL.");
                return;
            }

            val = UTIL.RemoveSingleQuotes(val);
            FQN = val;
        }

        public void setDb_id(ref string val)
        {
            if (Strings.Len(val) == 0)
            {
                val = "ECM.Library";
                // messagebox.show("SET: Field 'Db_id' cannot be NULL.")
                // Return
            }

            val = UTIL.RemoveSingleQuotes(val);
            DB_ID = val;
        }

        public void setVersionfiles(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            VersionFiles = val;
        }

        public void setCkmetadata(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            ckMetaData = val;
        }

        public void setCkpublic(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            ckPublic = val;
        }

        public void setCkdisabledir(ref string val)
        {
            val = UTIL.RemoveSingleQuotes(val);
            ckDisableDir = val;
        }



        // ** Generate the GET methods 
        public string getUserid()
        {
            if (Strings.Len(UserID) == 0)
            {
                MessageBox.Show("GET: Field 'Userid' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(UserID);
        }

        public string getIncludesubdirs()
        {
            if (IncludeSubDirs.ToUpper().Equals("TRUE"))
            {
                IncludeSubDirs = "Y";
            }

            if (IncludeSubDirs.ToUpper().Equals("FALSE"))
            {
                IncludeSubDirs = "N";
            }

            if (IncludeSubDirs.ToUpper().Equals("1"))
            {
                IncludeSubDirs = "Y";
            }

            if (IncludeSubDirs.ToUpper().Equals("0"))
            {
                IncludeSubDirs = "N";
            }

            return UTIL.RemoveSingleQuotes(IncludeSubDirs);
        }

        public string getFqn()
        {
            if (Strings.Len(FQN) == 0)
            {
                MessageBox.Show("GET: Field 'Fqn' cannot be NULL.");
                return "";
            }

            return UTIL.RemoveSingleQuotes(FQN);
        }

        public string getDb_id()
        {
            if (Strings.Len(DB_ID) == 0)
            {
                // messagebox.show("GET: Field 'Db_id' cannot be NULL.")
                return "ECM.Library";
            }

            return UTIL.RemoveSingleQuotes(DB_ID);
        }

        public string getVersionfiles()
        {
            if (VersionFiles.ToUpper().Equals("TRUE"))
            {
                VersionFiles = "Y";
            }

            if (VersionFiles.ToUpper().Equals("FALSE"))
            {
                VersionFiles = "N";
            }

            if (VersionFiles.ToUpper().Equals("1"))
            {
                VersionFiles = "Y";
            }

            if (VersionFiles.ToUpper().Equals("0"))
            {
                VersionFiles = "N";
            }

            return UTIL.RemoveSingleQuotes(VersionFiles);
        }

        public string getCkmetadata()
        {
            if (ckMetaData.ToUpper().Equals("TRUE"))
            {
                ckMetaData = "Y";
            }

            if (ckMetaData.ToUpper().Equals("FALSE"))
            {
                ckMetaData = "N";
            }

            if (ckMetaData.ToUpper().Equals("1"))
            {
                ckMetaData = "Y";
            }

            if (ckMetaData.ToUpper().Equals("0"))
            {
                ckMetaData = "N";
            }

            return UTIL.RemoveSingleQuotes(ckMetaData);
        }

        public string getCkpublic()
        {
            if (ckPublic.ToUpper().Equals("TRUE"))
            {
                ckPublic = "Y";
            }

            if (ckPublic.ToUpper().Equals("FALSE"))
            {
                ckPublic = "N";
            }

            if (ckPublic.ToUpper().Equals("1"))
            {
                ckPublic = "Y";
            }

            if (ckPublic.ToUpper().Equals("0"))
            {
                ckPublic = "N";
            }

            return UTIL.RemoveSingleQuotes(ckPublic);
        }

        public string getCkdisabledir()
        {
            return UTIL.RemoveSingleQuotes(ckDisableDir);
        }



        // ** Generate the Required Fields Validation method 
        public bool ValidateReqData()
        {
            if (UserID.Length == 0)
                return false;
            if (FQN.Length == 0)
                return false;
            if (DB_ID.Length == 0)
                return false;
            return true;
        }


        // ** Generate the Validation method 
        public bool ValidateData()
        {
            if (UserID.Length == 0)
                return false;
            if (FQN.Length == 0)
                return false;
            if (DB_ID.Length == 0)
                return false;
            return true;
        }


        // ** Generate the INSERT method 
        public bool Insert()
        {
            if (QuickRefEntry.Length == 0)
            {
                QuickRefEntry = "0";
            }

            if (OcrDirectory.Length == 0)
            {
                MessageBox.Show("OcrDirectory must be set to either Y or N");
            }

            FQN = FQN.Replace("'", "`");
            bool b = false;
            string s = "";
            s = s + " INSERT INTO Directory(";
            s = s + "UserID,";
            s = s + "IncludeSubDirs,";
            s = s + "FQN,";
            s = s + "DB_ID,";
            s = s + "VersionFiles,";
            s = s + "ckMetaData,";
            s = s + "ckPublic,";
            s = s + "ckDisableDir, QuickRefEntry, OcrDirectory, OcrPdf, ArchiveSkipBit, DeleteOnArchive) values (";
            s = s + "'" + UserID + "'" + ",";
            s = s + "'" + IncludeSubDirs + "'" + ",";
            s = s + "'" + FQN + "'" + ",";
            s = s + "'" + DB_ID + "'" + ",";
            s = s + "'" + VersionFiles + "'" + ",";
            s = s + "'" + ckMetaData + "'" + ",";
            s = s + "'" + ckPublic + "'" + ",";
            s = s + "'" + ckDisableDir + "'" + ",";
            s = s + "'" + QuickRefEntry + "'" + ",";
            s = s + "'" + OcrDirectory + "'" + ",";
            s = s + "'" + OcrPdf + "'" + ",";
            if (ArchiveBit == true)
            {
                s = s + "1" + ",";
            }
            else
            {
                s = s + "0" + ",";
            }

            s = s + "'" + DeleteOnArchive + "')";
            return DBARCH.ExecuteSqlNewConn(s, false);
        }


        // ** Generate the UPDATE method 
        public bool Update(string WhereClause, string OcrDirectory)
        {
            bool b = false;
            string s = "";
            if (OcrDirectory.Length == 0)
            {
                MessageBox.Show("OcrDirectory must be set to either Y or N");
            }

            if (Strings.Len(WhereClause) == 0)
                return false;
            string ckDisableDir = getCkdisabledir();
            if (ckDisableDir.ToUpper() == "TRUE")
            {
                ckDisableDir = "Y";
            }

            if (ckDisableDir.ToUpper() == "FALSE")
            {
                ckDisableDir = "N";
            }

            if (ckDisableDir.ToUpper() == "1")
            {
                ckDisableDir = "Y";
            }

            if (ckDisableDir.ToUpper() == "0")
            {
                ckDisableDir = "N";
            }

            s = "";
            s = s + " update Directory set ";
            s = s + "UserID = '" + getUserid() + "'" + ", ";
            s = s + "IncludeSubDirs = '" + getIncludesubdirs() + "'" + ", ";
            s = s + "FQN = '" + getFqn() + "'" + ", ";
            s = s + "VersionFiles = '" + getVersionfiles() + "'" + ", ";
            s = s + "ckMetaData = '" + getCkmetadata() + "'" + ", ";
            s = s + "ckPublic = '" + getCkpublic() + "'" + ", ";
            s = s + "ckDisableDir = '" + ckDisableDir + "',";
            s = s + "OcrDirectory = '" + OcrDirectory + "',";
            s = s + "OcrPdf = '" + OcrPdf + "',";
            if (ArchiveBit == true)
            {
                s = s + "ArchiveSkipBit = 1, ";
            }
            else
            {
                s = s + "ArchiveSkipBit = 0, ";
            }

            s = s + "DeleteOnArchive = '" + DeleteOnArchive + "'";
            WhereClause = " " + WhereClause;
            s = s + WhereClause;
            return DBARCH.ExecuteSqlNewConn(s, false);
        }


        // ** Generate the SELECT method 
        public SqlDataReader SelectRecs()
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "UserID,";
            s = s + "IncludeSubDirs,";
            s = s + "FQN,";
            s = s + "DB_ID,";
            s = s + "VersionFiles,";
            s = s + "ckMetaData,";
            s = s + "ckPublic,";
            s = s + "ckDisableDir ";
            s = s + " FROM Directory";
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }


        // ** Generate the Select One Row method 
        public SqlDataReader SelectOne(string WhereClause)
        {
            bool b = false;
            string s = "";
            var rsData = default(SqlDataReader);
            s = s + " SELECT ";
            s = s + "UserID,";
            s = s + "IncludeSubDirs,";
            s = s + "FQN,";
            s = s + "DB_ID,";
            s = s + "VersionFiles,";
            s = s + "ckMetaData,";
            s = s + "ckPublic,";
            s = s + "ckDisableDir ";
            s = s + " FROM Directory";
            s = s + WhereClause;
            string CS = DBARCH.setConnStr();     // DBARCH.getGateWayConnStr(gGateWayID) : Dim CONN As New SqlConnection(CS) : CONN.Open() : Dim command As New SqlCommand(s, CONN) : rsData = command.ExecuteReader()
            return rsData;
        }


        // ** Generate the DELETE method 
        public bool Delete(string WhereClause)
        {
            bool b = false;
            string s = "";
            if (Strings.Len(WhereClause) == 0)
                return false;
            WhereClause = " " + WhereClause;
            s = " Delete from Directory";
            s = s + WhereClause;
            b = DBARCH.ExecuteSqlNewConn(s, false);
            return b;
        }


        // ** Generate the Zeroize Table method 
        public bool Zeroize()
        {
            bool b = false;
            string s = "";
            s = s + " Delete from Directory";
            b = DBARCH.ExecuteSqlNewConn(s, false);
            return b;
        }


        // ** Generate Index Queries 
        public int cnt_PKII2(string FQN, string UserID)
        {
            int B = 0;
            string TBL = "Directory";
            string WC = "Where FQN = '" + FQN + "' and   UserID = '" + UserID + "'";
            B = DBARCH.iGetRowCount(TBL, WC);
            return B;
        }     // ** cnt_PKII2

        // ** Generate Index ROW Queries 
        public SqlDataReader getRow_PKII2(string FQN, string UserID)
        {
            SqlDataReader rsData = null;
            string TBL = "Directory";
            string WC = "Where FQN = '" + FQN + "' and   UserID = '" + UserID + "'";
            rsData = DBARCH.GetRowByKey(TBL, WC);
            if (rsData.HasRows)
            {
                return rsData;
            }
            else
            {
                return null;
            }
        }     // ** getRow_PKII2

        /// Build Index Where Caluses
        public string wc_PKII2(string FQN, string UserID)
        {
            string WC = "Where FQN = '" + FQN + "' and   UserID = '" + UserID + "'";
            return WC;
        }     // ** wc_PKII2
    }
}